Resolving GridUI caption error

ganimtron-10 opened the pull request:
Fixing Bug mentioned at #463.

skoudoro reviewed (commented):
Hi @ganimtron-10,

Thank you for doing that! Could you add a test also? Thank you

@antrikshmisri, Can you test this PR and check if it fix the reported issue. Thanks!

ganimtron-10 commented:
I have added tests for the changes.

skoudoro commented:
Thank for the update @ganimtron-10!

ping @antrikshmisri

antrikshmisri commented on fury/tests/test_actors.py:
```diff
@@ -1176,6 +1176,7 @@ def timer_callback(_obj, _event):
     arr[arr < 20] = 0
     report = window.analyze_snapshot(arr)
     npt.assert_equal(report.objects, 6)
+    npt.assert_equal(type(container._items[0]), actor.Container)
```
Please avoid using private members, instead use the property.

ganimtron-10 replied:
Done!

antrikshmisri commented on fury/tests/test_actors.py:
```diff
@@ -1196,6 +1197,7 @@ def timer_callback(_obj, _event):
     arr = window.snapshot(scene)
     report = window.analyze_snapshot(arr)
     npt.assert_equal(report.objects > 6, True)
+    npt.assert_equal(type(container._items[0]), actor.Container)
```
Same comment as above.

antrikshmisri reviewed (commented):
Hey @ganimtron-10, please see my comments below. Rest everything seems to be working fine. Thanks!

ganimtron-10 commented:
Thanks @antrikshmisri for the review!
I have made changes according to the comments.

antrikshmisri commented:
Hey @ganimtron-10, thanks for the update. LGTM.

ganimtron-10 commented:
Hii @antrikshmisri !! 
Is there any thing left to be changed in this pr??

skoudoro reviewed (commented):
Thanks @antrikshmisri  for the review. 

I need to test this PR and we can go ahead and merge it. It should be done by the end of this week.

ganimtron-10 commented:
Hello @skoudoro ,
Any updates on this?

skoudoro reviewed (changes requested):
Hi @ganimtron-10,

I do not think this is the right approach. 

It should be fixed at `GridUI` class level and not at the `grid`function level. 

Currently, you are adding complexity which is not necessary. to be more precise, you are doing:  `Container(Container(actor), Container(actor), etc...)`  and each container has only one actor so useless.

Please, let me know if you have time to update this PR, otherwise, I will close it. 

Thanks!

ganimtron-10 commented:
Hello @skoudoro ,
Thanks for the update!

Last time, I just got confused between the return type of the `grid` function and the type of the `items` as I thought the `grid` function is inconsistent (ie. which returns either `Container` or `Actor`) due to which I implemented that approach, but this isn't the case.

Now I have updated the PR and made changes into the `GridUI` itself.
PTAL.

skoudoro reviewed (commented):
Much better! Thank you for the quick update! 

Can you just update the test of `gridUI`?  Then, it will be ready to merge. You could start with the initial issue #463 and make sure that we have the expected output in the test. 

Thanks !

skoudoro commented:
@ganimtron-10, this PR is almost ready to be merged. 

See my comment above. Can you update the tests? Thank you!

ganimtron-10 commented:
Hello @skoudoro ,
I tried creating a similar test and as per the older conversation used try and except block to test whether it throw error or not. Please let me know if its correct or not.

ganimtron-10 commented:
Hey @skoudoro ,
I have updated the tests and they are now passing on `macOS`.

skoudoro commented:
Thank you for this @ganimtron-10, looks good!
